.github/workflows/compare_upstream.yml

```diff
@@ -0,0 +1,35 @@
+name: Compare Upstream
+
+on:
+  schedule: 
+   - cron: "0 1 * * *"
+
+jobs:
+  compare:
+    runs-on: ubuntu-latest
+    outputs:
+      upstream_change: ${{ steps.compare_step.outputs.upstream_change }}
+
+    steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v3
+    - name: Set up Python 3.11
+      uses: actions/setup-python@v3
+      with:
+        python-version: "3.11"
+    - name: Upgrade pip and install mpc_obscodes
+      run: python -m pip install --upgrade build requests mpc-obscodes
```

Comment:
We may encounter problems both checking out the package using `actions/checkout` and then pip installing the latest version from PyPI. 

Might I recommend using pip download and doing something like this instead:

```
# Download the contents of the PyPI package to the `remote` folder
pip download --no-binary :all: -d remote mpc-obscodes

# Extract the file we want to md5 compare and store it in `existing/obscodes_extended.json`
# You may need to adjust this command if running on macOS
tar -zxf remote/* --wildcards --no-anchored '*obscodes_extended.json' -O > existing/obscodes_extended.json

# Fetch the new files we care about and stick them in `new`
python mpc_obscodes/fetch.py --compressed_file new/obscodes_extended.json.gz --decompressed_file new/obscodes_extended.json --md5_file new/obscodes_extended_new.md5

# Compare the md5s of the two files and indicate upstream change in the output
if [ "$(md5sum existing/obscodes_extended.json new/obscodes_extended.json | awk '{print $1}' | uniq -c | grep -c "1 ")" -eq 1 ]; then export GITHUB_OUTPUT+="upstream_change=true "; fi
```